### ISSUE TITLE
pred_wh not finite

### ISSUE THREAD

**Issue by hank-sunday** [duwt/OrienMask#4]
I follow the install and train steps, but when I train to some pics ,there is a error "pred_wh not finite". 
how can I figure it out?

**Comment by duwt** [comment]
Sorry about that. Increasing the learning rate warm-up steps from 500 to 1000 makes the training procedure more stable. Please refer to L252 as below.
https://github.com/duwt/OrienMask/blob/fce422f546f7ac83305208d98a8cef590902018f/config/base.py#L248-L256

**Comment by hank-sunday** [comment]
thanks a lot!

**Comment by hank-sunday** [comment]
No matter what I set the warmup_iter to 1000, or I change the warmup_ratio and/or gamma to 0.05. the lr will increase slowly ,after it reach 1.00e-03 ,the loss will explode (larger than 2000) and "pred_wh not finite" will be reported.

I use 1 GPU and torch 1.9.0+cu111

How can i do? thanks.

**Comment by duwt** [comment]
I try to run the script three times but do not meet the error before 5000 steps. If you use a single GPU for training, `accumulate` should be 2 instead of 1 as mentioned in the readme. Please refer to L8 as below.
https://github.com/duwt/OrienMask/blob/fce422f546f7ac83305208d98a8cef590902018f/config/config_train.py#L3-L25

**Comment by aoko-sys** [comment]
@hank-sunday Hello! I had the same problem as you. Did you solve the problem?

**Comment by aswa123** [comment]
@hank-sunday , @aoko-sys @duwt 
Hello, 
same problem as yours.
but on different iterations.

**Comment by duwt** [comment]
Single GPU configuration(n_gpu=1, batch_size=8, accumulate=2) seems unstable in our recent experiments. It's recommended to use n_gpu=2, batch_size=8, accumulate=1. We are truly sorry about that. In our earlier experiments, it's ok but we have no exact environment settings about that.

Another option is to reduce the orientation loss weight from 20 to 15 or even 10, which may cause some accuracy drop.

**Comment by pupu-chenyanyan** [comment]
@hank-sunday @aswa123 @aoko-sys  Hello,  two  gpus also met the problem.Did you solve the problem?
